x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -265,6 +266,63 @@ export function registerIndexRoutes({
     })
   );
 
+  router.post(
+    {
+      path: '/internal/enterprise_search/indices/{indexName}/ml_inference/pipeline_processors',
+      validate: {
+        params: schema.object({
+          indexName: schema.string(),
+        }),
+        body: schema.object({
+          pipeline_name: schema.string(),
+          model_id: schema.string(),
+          source_field: schema.string(),
+          destination_field: schema.maybe(schema.nullable(schema.string())),
+        }),
+      },
+    },
+
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { client } = (await context.core).elasticsearch;
+      const {
+        model_id: modelId,
+        pipeline_name: pipelineName,
+        source_field: sourceField,
+        destination_field: destinationField,
+      } = request.body;
+
+      try {
+        await createMlInferencePipeline(
+          pipelineName,
+          modelId,
+          sourceField,
+          destinationField || modelId,
+          client.asCurrentUser
+        );
```

Comment:
Heads up - I'm adding a new step here that does the following:
- Fetches the `<index-name>@ml-inference` pipeline
  - TBD if we need to backfill it just-in-time if it's missing
- Adds the new sub-pipeline to the list of processors of `<index-name>@ml-inference`
- Saves it

Reply:
> backfill it just-in-time if it's missing

This _feels_ right. And this also reduces our 100% dependency on connector pipeline support to ... close to 0%, maybe?

Reply:
I had a chat with @TattdCodeMonkey and agreed that we won't backfill here, as it's an inconsistent scenario (see [reason](https://github.com/elastic/kibana/pull/140645/files#r972173992)). We can add it as a follow-up if necessary.